### PR TITLE
Spelling corrections

### PR DESCRIPTION
**Pull request by albert-github**
A number of spelling corrections as reported by the codespell (see: https://github.com/codespell-project/codespell) program and lists.

Some remarks:
- not considered are grammatical errors
- not considered are names in the code although there are a number that could be improved (like `childs` -> `children`)
- in the documentation it could be made clearer what are variables and what is running text (e.g. by placing variables in running text between backticks)
- some comments are in the French language, I think it would be better to have them in the English (United States version).

### CODE REVIEW

File: src/Persistent_cohomology/example/rips_multifield_persistence.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ void program_options(int argc, char * argv[]
       ("min-field-charac,p", po::value<int>(&min_p)->default_value(2),
        "Minimal characteristic p of the coefficient field Z/pZ.")
       ("max-field-charac,q", po::value<int>(&max_p)->default_value(1223),
-       "Minimial characteristic q of the coefficient field Z/pZ.")
+       "Minimal characteristic q of the coefficient field Z/pZ.")
```
Comment (mglisse) [inline src/Persistent_cohomology/example/rips_multifield_persistence.cpp:107]:
```suggestion
       "Maximal characteristic q of the coefficient field Z/pZ.")
```
(the mix of p and q also looks strange, but that's a different problem)

File: src/python/test/test_simplex_tree.py [hunk 2]
```diff
@@ -528,7 +528,7 @@ def test_expansion_with_blocker():
 
     def blocker(simplex):
         try:
-            # Block all simplices that countains vertex 6
+            # Block all simplices that contains vertex 6
```
Comment (mglisse) [inline src/python/test/test_simplex_tree.py:531]:
```suggestion
            # Block all simplices that contain vertex 6
```

File: src/Bitmap_cubical_complex/include/gudhi/Bitmap_cubical_complex.h [hunk 3]
```diff
@@ -237,7 +237,7 @@ class Bitmap_cubical_complex : public T {
    * Filtration_simplex_iterator class provides an iterator though the whole structure in the order of filtration.
    * Secondary criteria for filtration are:
    * (1) Dimension of a cube (lower dimensional comes first).
-   * (2) Position in the data structure (the ones that are earlies in the data structure comes first).
+   * (2) Position in the data structure (the ones that are earliest in the data structure comes first).
```
Comment (mglisse) [inline src/Bitmap_cubical_complex/include/gudhi/Bitmap_cubical_complex.h:240]:
```suggestion
   * (2) Position in the data structure (the ones that are earliest in the data structure come first).
```

**Review by mglisse — COMMENTED** [review]
ok with those minor changes, thanks.

**Comment by albert-github** [comment]
Thanks for the feedback.

**Review by VincentRouvreau — APPROVED** [review]

**Pull request merged**